### ISSUE TITLE
$ref keys in input are not escaped

### ISSUE THREAD

**Issue by maximilianMairinger** [YChebotaev/json-decycle#6]
Using the following input

```js
const tom = {
  name: "Tom",
  age: 24
}

const emma = {
  name: "Emma",
  age: 23,
  test: {$ref: "#/invalid"} //!
}

tom.likes = emma
emma.likes = tom

const input = { tom }
```

Produces an incorrect result on serialization

```js
const json = JSON.stringify(input, decycle())
const obj = JSON.parse(json, retrocycle())

if (obj.tom.loves.test === undefined) throw new Error('Should be {$ref: "#/invalid"}')
```

**Comment by maximilianMairinger** [comment]
Also invalid refs (like below) should be considered and are in #7 

```js
const input = {$ref: "/invalid"}
```

**Comment by bminer** [comment]
I am probably being dumb, but I don't understand the issue here.  Since the ref is invalid, why should it be serialized in the JSON?

**Comment by maximilianMairinger** [comment]
TLDR: Object keys may be subject to user input, thus $ref may be used unintentionally. In this case, the behavior is not defined (value of type string causes undefined; number throws). 

Although it isn't necessarily a security issue, it doesn't comply with the (intuitive) "same in as out" convention, which this library should confirm to imo. You may disclose in the doc that $ref is a forbidden key and may not be used (when expecting the convention mentioned above). Hence, the library user is obligated to a: escape $ref keys, or b: make sure that object keys are not subject to user input.

At the very least, this case should be defined (in the docs), similar to how JSON doesn't allow circularity. But I prefer the proposed method, as it is less restricting.

**Comment by bminer** [comment]
I feel like it's an application-level issue to escape keys equal to `$ref` to avoid this behavior.  I'd be happy to hear arguments to the contrary.  Closing this issue for now.
